### DISCUSSION TITLE
rstudioapi shims not working

### DISCUSSION THREAD

**Discussion by olivroy** [posit-dev/positron#4970] [Q&A] [answered]
IIUC from https://github.com/posit-dev/positron/issues/2605, many rstudioapi functions should work.

However, I am unable to access any of them.

For example, running `rstudioapi::getSourceEditorContext()` in the Positron R console yields an error message RStudio not running.

I am using the User Setup on Windows (latest release)

I wonder if I understood something wrong, or if there is a bug somewhere.

**Comment by juliasilge** [comment]
I am able to run `rstudioapi::getSourceEditorContext()` on both Mac and Windows without any problems with Positron 2024.10.0 build 14:

![Screenshot 2024-10-09 at 2 48 05 PM](https://github.com/user-attachments/assets/d937dc9f-6ab6-4c18-93fb-dd0b8b6b4b53)

Can you share a bit more about what you are seeing? What happens if you run `.rs.api.getSourceEditorContext()` directly in Positron?

**Reply by olivroy** [reply]
Thanks for taking a look. `.rs.api.getSourceEditorContext()` works. But `rstudioapi::getSourceEditorContext()` errors like so:
![image of failing code](https://github.com/user-attachments/assets/b3f52436-4288-4b6b-85f3-7386f1fdb29d)

**Reply by juliasilge** [reply]
Just to confirm, `rstudioapi::isAvailable()` returns `FALSE` for you in Positron?

**Reply by olivroy** [reply]
Indeed!

**Reply by juliasilge** [reply, marked as answer]
I opened https://github.com/posit-dev/positron/issues/4976 to track and prioritize this bug. Thank you again!

**Reply by olivroy** [reply]
Thanks! Will close discussion
